ocaml/xapi/daemon_manager.ml

```diff
@@ -63,14 +63,29 @@ module Make(D : DAEMON) = struct
 		end
 	| Function f -> f ()
 
-	let with_daemon_stopped f =
+	let start = D.start
+
+	let stop ?timeout () =
+		match timeout with
+		| Some t -> begin
```

Comment:
This begin end is for the `try` right? So that the other branch of the match doesn't get into the exception pattern match. Can we place the begin end around there instead?